python/GangaLHCb/Lib/GaudiRunApp/PythonOptsCmakeParser.py

```diff
@@ -0,0 +1,153 @@
+import os.path
+import tempfile
+from Ganga.GPIDev.Lib.File import LocalFile
+import Ganga.Utility.logging
+from GangaLHCb.Lib.LHCbDataset import LHCbDataset
+from Ganga.Core import ApplicationConfigurationError
+from Ganga.Utility.files import expandfilename
+logger = Ganga.Utility.logging.getLogger()
+
+def readInputData(optsfiles, app):
+    '''Returns a LHCbDataSet object from a list of options files. The
+       optional argument extraopts will decide if the extraopts string inside
+       the application is considered or not.
+
+       Usage examples:
+       # Create an LHCbDataset object with the data found in the optionsfile
+       l=DaVinci(version='v22r0p2').readInputData([\"~/cmtuser/DaVinci_v22r0p2/Tutorial/Analysis/options/Bs2JpsiPhi2008.py\"])
+       # Get the data from an options file and assign it to the jobs inputdata field
+       j.inputdata = j.application.readInputData([\"~/cmtuser/DaVinci_v22r0p2/Tutorial/Analysis/options/Bs2JpsiPhi2008.py\"])
+
+    '''
+    if not isinstance(optsfiles, []):
```

Comment:
I'm not sure this does what you expect. This will just fail and raise an exception. Is this function even called anywhere?